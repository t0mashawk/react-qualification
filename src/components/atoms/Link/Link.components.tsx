import React from 'react'

interface LinkProps{
    text:string,
    href:string
}

function Link(props:LinkProps){
    const {text, href} = props
    return(
        <a href={href}>{text}</a>
    )
}

export default Link