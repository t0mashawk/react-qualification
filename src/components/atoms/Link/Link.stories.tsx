import React from "react";
import { storiesOf } from "@storybook/react";
import Link from "./Link.components"

storiesOf("Link", module)
    .add("default",
        () => <Link text="link" href="./"/>
    )