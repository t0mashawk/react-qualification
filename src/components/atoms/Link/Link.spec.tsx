import React from "react";
import renderer from "react-test-renderer"

import Link from "./Link.components"

test("Link renders as expected", () => {
  const component = renderer.create(
    <Link text="Example Value" href="www.google.com"/>
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});