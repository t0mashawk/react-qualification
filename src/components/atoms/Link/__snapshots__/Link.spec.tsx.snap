// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders as expected 1`] = `
<a
  href="www.google.com"
>
  Example Value
</a>
`;
